timemachine/md/minimizer.py

```diff
@@ -26,6 +28,15 @@ class MinimizationError(Exception):
     pass
 
 
+def check_force_norm(du_dx, threshold=MAX_FORCE_NORM):
+    """raise Minimization error if the force on any atom exceeds MAX_FORCE_NORM"""
+    norm = np.linalg.norm(du_dx, axis=-1)
+    if not np.all(norm < threshold):
+        raise MinimizationError(
+            f"Minimization failed to reduce large forces below threshold: |frc| = {norm} > {threshold}"
```

Comment:
nit: for readability might be nice to either do the max norm or sort them. When the number of particles is large you may not be able to see the force that is large.

On line 415 in the previous it would print the last.

Reply:
Ahh, good call -- this was derived from the version in `minimize_host_4d` https://github.com/proteneer/timemachine/blob/195ce1c0f9cebc7f184b5322dac2faba0345ef7e/timemachine/md/minimizer.py#L175-L179
but the version in `local_minimize` is more informative https://github.com/proteneer/timemachine/blob/195ce1c0f9cebc7f184b5322dac2faba0345ef7e/timemachine/md/minimizer.py#L414-L419

Reply:
Expanded error message in https://github.com/proteneer/timemachine/pull/978/commits/c516cbfec71782ad70267017cd2cf0a4d41eb752 -- report max force norm, as well as the number of atoms that exceed threshold